awesome_owl/static/src/todo/todoList.js

```diff
@@ -0,0 +1,33 @@
+import { Component, useState } from "@odoo/owl"
+import { useAutofocus } from "@web/core/utils/hooks";
+import { Todo } from "./todo"
+
+export class TodoList extends Component {
+    static template = "awesome_owl.TodoList"
+    static components = { Todo }
+    numAllocatedIds = 0;
+
+    setup() {
+        this.todoAddInputRef = useAutofocus({ refName: 'todoAddInputRef' });
+        this.state = useState({
+            todos: [],
+        });
+    };
+
+    addTodo(event) {
+        if (event.keyCode === 13 && event.target.value !== "") {
+            this.numAllocatedIds++;
+            const newTodo = { description: event.target.value, id: this.numAllocatedIds, isCompleted: false };
+            this.state.todos.push(newTodo);
```

Comment:
Is it necessary to use a temporary constant here?
```suggestion
            this.state.todos.push({ description: event.target.value, id: this.numAllocatedIds, isCompleted: false });
```